common/hyperscaler/rules/definitions.go

```diff
@@ -0,0 +1,164 @@
+package rules
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/broker"
+)
+
+const (
+	PR_ATTR_NAME     = "PR"
+	HR_ATTR_NAME     = "HR"
+	EU_ATTR_NAME     = "EU"
+	SHARED_ATTR_NAME = "S"
+
+	HYPERSCALER_LABEL = "hyperscalerType"
+	EUACCESS_LABEL    = "euAccess"
+	SHARED_LABEL      = "shared"
+
+	ASTERISK                 = "*"
+	ATTRIBUTE_WITH_VALUE     = "attr"
+	SIGNATURE_ATTR_SEPARATOR = ":"
+)
+
+var InputAttributes = []Attribute{
+	{
+		Name:            PR_ATTR_NAME,
+		Description:     "Platform Region",
+		Setter:          setPlatformRegion,
+		Getter:          func(r *Rule) string { return r.PlatformRegion },
+		MatchableGetter: func(r *ProvisioningAttributes) string { return r.PlatformRegion },
+		input:           true,
+		output:          false,
+		HasValue:        true,
+		ApplyLabel: func(r *Rule, labels map[string]string) map[string]string {
+			labels[HYPERSCALER_LABEL] += "_" + r.PlatformRegion
+			return labels
+		},
+	},
+	{
+		Name:        HR_ATTR_NAME,
+		Description: "Hyperscaler Region",
+		Setter:      setHyperscalerRegion,
+		Getter:      func(r *Rule) string { return r.HyperscalerRegion },
+		input:       true,
+		output:      false,
+		HasValue:    true,
+		ApplyLabel: func(r *Rule, labels map[string]string) map[string]string {
+			labels[HYPERSCALER_LABEL] += "_" + r.HyperscalerRegion
+			return labels
+		},
+	},
+}
+
+var OutputAttributes = []Attribute{
+	{
+		Name:        EU_ATTR_NAME,
+		Description: "EU Access",
+		Setter:      setEuAccess,
+		Getter:      func(r *Rule) string { return strconv.FormatBool(r.EuAccess) },
+		input:       false,
+		output:      true,
+		HasValue:    false,
+		ApplyLabel: func(r *Rule, labels map[string]string) map[string]string {
+			if r.EuAccess {
+				labels[EUACCESS_LABEL] = "true"
+			}
+			return labels
+		},
+	},
+	{
+		Name:        SHARED_ATTR_NAME,
+		Description: "Shared",
+		Setter:      setShared,
+		Getter:      func(r *Rule) string { return strconv.FormatBool(r.Shared) },
+		input:       false,
+		output:      true,
+		HasValue:    false,
+		ApplyLabel: func(r *Rule, labels map[string]string) map[string]string {
+			if r.Shared {
+				labels[SHARED_LABEL] = "true"
+			}
+
+			return labels
+		},
+	},
+}
+
+var AllAttributes = append(InputAttributes, OutputAttributes...)
+
+func setShared(r *Rule, value string) (*Rule, error) {
+	if r.Shared {
+		return nil, fmt.Errorf("Shared already set")
+	}
+
+	r.ContainsOutputAttributes = true
+	r.Shared = true
+
+	return r, nil
+}
+
+func setEuAccess(r *Rule, value string) (*Rule, error) {
```

Comment:
value is not used

Reply:
Yes but it has to comply with the interface of the Attribute:
![image](https://github.com/user-attachments/assets/193eeef0-4a20-4227-a66e-808640615e89)

Reply:
We should consider how to redesign it.

Reply:
Feel free to change implementation however you want.